server/src/test/java/io/crate/integrationtests/InformationSchemaTest.java

```diff
@@ -638,47 +647,101 @@ public void testSelectFromTableColumns() {
+        assertEquals("stuff", response.rows()[7][cols.get("column_name")]);
+        assertEquals("stuff['level1']", response.rows()[8][cols.get("column_name")]);
+        assertEquals("stuff['level1']['level2']", response.rows()[9][cols.get("column_name")]);
+        assertEquals("stuff['level1']['level2_nullable']", response.rows()[10][cols.get("column_name")]);
+
+        assertEquals("integer", response.rows()[0][cols.get("data_type")]);
+        assertEquals(null, response.rows()[0][cols.get("datetime_precision")]);
+        assertEquals("NEVER", response.rows()[0][cols.get("is_generated")]);
+        assertEquals(32, response.rows()[0][cols.get("numeric_precision")]);
+        assertEquals(2, response.rows()[0][cols.get("numeric_precision_radix")]);
+        assertEquals(3, response.rows()[0][cols.get("ordinal_position")]);
+        assertEquals(defaultSchema, response.rows()[0][cols.get("table_catalog")]);
+        assertEquals("test", response.rows()[0][cols.get("table_name")]);
+
+        assertEquals(false, response.rows()[0][cols.get("is_identity")]);
+        assertEquals(null, response.rows()[0][cols.get("identity_generation")]);
+        assertEquals(null, response.rows()[0][cols.get("identity_start")]);
+        assertEquals(null, response.rows()[0][cols.get("identity_increment")]);
+        assertEquals(null, response.rows()[0][cols.get("identity_maximum")]);
+        assertEquals(null, response.rows()[0][cols.get("identity_minimum")]);
+        assertEquals(null, response.rows()[0][cols.get("identity_cycle")]);
+
+        assertEquals(false, response.rows()[0][cols.get("is_nullable")]);
+        assertEquals(false, response.rows()[2][cols.get("is_nullable")]);
+        assertEquals(true, response.rows()[3][cols.get("is_nullable")]);
+        assertEquals(false, response.rows()[7][cols.get("is_nullable")]);
+        assertEquals(false, response.rows()[8][cols.get("is_nullable")]);
+        assertEquals(false, response.rows()[9][cols.get("is_nullable")]);
+        assertEquals(true, response.rows()[10][cols.get("is_nullable")]);
+
+        assertEquals(8, response.rows()[1][cols.get("numeric_precision")]);
+        assertEquals(16, response.rows()[6][cols.get("numeric_precision")]);
+        assertEquals(53, response.rows()[4][cols.get("numeric_precision")]);
+        assertEquals(24, response.rows()[5][cols.get("numeric_precision")]);
+
+        assertEquals(Map.of("name","stuff","path", List.of()) , response.rows()[7][cols.get("column_details")]); // column_details
+        assertEquals(Map.of("name","stuff","path", List.of("level1")) , response.rows()[8][cols.get("column_details")]); // column_details
+        assertEquals(Map.of("name","stuff","path", List.of("level1","level2")) , response.rows()[9][cols.get("column_details")]); // column_details
+        assertEquals(Map.of("name","stuff","path", List.of("level1","level2_nullable")) , response.rows()[10][cols.get("column_details")]); // column_details
```

Comment:
Can't we use `printedTable` here instead of checking columns and later on values?

Reply:
I first started out with `printedTable`, however it would be a very big table with lots of `null` values, making it almost impossible to find deviations in test cases, which then probably would lead to generating the output automatically, which would defeat the test case purpose. I rather wondered if adding a way to access fields using column-names should be added to the test suite.  With 42 columns this table is rather wide.

Reply:
>  making it almost impossible to find deviations in test cases

Normally, diff works quite well on `printedTable` outputs. And yes the output asserted here would be taken from a manual, verified run.  Readability will improve a lot when using `printedTable` and also this way of asserting columns requires a lot of manual refactoring once a new column is added.

Reply:
the output of printed table does not include column names.
If I add another column to the table, one needs to manually check (cumbersome) or generate a test output (which would prevent us from ensuring that the former test case is still valid)

While adding a new test case for a new column with the key approach is just adding a new line.

```
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=age, path=[]}| age| integer| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| false| 32| 2| NULL| 3| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=b, path=[]}| b| char| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| true| 8| 2| NULL| 4| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=col1, path=[]}| col1| integer| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| false| 32| 2| NULL| 1| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=col2, path=[]}| col2| text| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| true| NULL| NULL| NULL| 2| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=d, path=[]}| d| double precision| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| true| 53| 2| NULL| 6| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=f, path=[]}| f| real| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| true| 24| 2| NULL| 7| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=s, path=[]}| s| smallint| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| true| 16| 2| NULL| 5| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=stuff, path=[]}| stuff| object| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| false| NULL| NULL| NULL| 8| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=stuff, path=[level1]}| stuff['level1']| object| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| false| NULL| NULL| NULL| 9| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=stuff, path=[level1, level2]}| stuff['level1']['level2']| text| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| false| NULL| NULL| NULL| 10| derj| test| derj| NULL| NULL| NULL
NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| {name=stuff, path=[level1, level2_nullable]}| stuff['level1']['level2_nullable']| text| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NULL| NEVER| false| true| NULL| NULL| NULL| 11| derj| test| derj| NULL| NULL| NULL
```

Reply:
I tend to agree with @proddata on this, If I'm not mistaken, that was the reason that those assertions were written like this (without the usage of `printedTable`) in the first place.

Reply:
> the output of printed table does not include column names.

right, my fault.

>  If I'm not mistaken, that was the reason that those assertions were written like this (without the usage of printedTable) in the first place.

I don't think that there was a specific reason.

> While adding a new test case for a new column with the key approach is just adding a new line.